### ISSUE TITLE
MKL install dependencies error in M1 Mac

### ISSUE THREAD

**Issue by SeaCelo** [EAPD-DRB/OG-ZAF#3]
_Note: This problem occurs on a M1 Mac that uses miniforge instead of standard Anaconda installation. Numpy installs openblas instead of mkl for acceleration (the former is ARM native)._ 

The command `conda env create -f environment.yml` results in an installation error: 
```
ResolvePackageNotFound: 
  - mkl[version='>=2020.2']
```
It is possible to comment out line 6 of environment.yml to get past this error. However, the error returns on line 34 during the pip installation of ogcore, which also has MKL as a dependency.

To avoid having to install MKL and run it under rosetta emulation, we need to ignore the MKL dependency.

Suggested (manual) solution:

1. comment out lines 6 and 34 of environment.yml. 
2. after running the `conda env create -f environment.yml` step, run: 

- `conda activate ogzaf-dev`
- `pip install --no-dependencies ogcore` 
- continue with the installation process: `pip install -e .`

Using these steps I successfully installed OG-ZAF and ran the test `python run_og_zaf.py`. 
Baseline run time: 1116.8 seconds (18.6 minutes)
Reform run time: 1092.3 (18.2 minutes)
Total run time: 2209.1 seconds (36.8 minutes)

**Comment by rickecon** [comment]
@SeaCelo. I suspect that the issue is the miniforge installation of Python, although I don't know why the `mkl` requirement would break with miniforge. All of our testing is with the Anaconda distribution of Python on our local machines and the miniconda distribution of Python for our GitHub Action CI testing. We don't have this problem with the miniconda distribution of Python on our cloud GH CI tests. See this Stackoverflow thread on [the difference between miniforge and miniconda](https://stackoverflow.com/questions/60532678/what-is-the-difference-between-miniconda-and-miniforge). Here are two suggestions.
1. You could install the [miniconda Python distribution](https://docs.conda.io/en/latest/miniconda.html) on your local machine instead of the miniforge distribution.
2. You could use the [Anaconda distribution](https://www.anaconda.com/products/distribution) of Python on your local machine instead of the miniforge distribution.

The other option is for us to figure out why the `mkl` package is creating an incompatibility with the `environment.yml` with miniforge. I would like to avoid this work in the short run if we can make it work with miniconda and Anaconda.

**Comment by SeaCelo** [comment]
@rickecon I agree and this is not worth making any significant revisions to the installation process. Maybe we can close this issue and just refer anyone who wants to use miniforge to this thread.

**Comment by rickecon** [comment]
@SeaCelo. Have you tried miniconda instead of miniforge? I would be interested to know if that works for you. Let's leave this issue open for now.

**Comment by SeaCelo** [comment]
@rickecon I'd rather not switch distributions if I can avoid it. I trust that it would work with miniconda as your tests suggest, but I have other projects and OG-CORE and OG-ZAF are running well in my current setup with the small installation adjustments. 

I am curious what the performance difference would be. From what I can tell, mkl has still not been ported to ARM, so it would run under Rosetta in my system. There are some tests of numpy that show a significant performance hit with using mkl on an ARM Mac. [This comment compares the performance of miniconda and miniforge using python](https://developer.apple.com/forums/thread/695963?answerId=697568022#697568022). This seems to argue strongly for avoiding Rosetta where possible. 

I may be reading this all wrong, however. I'd be happy to devote some time to removing miniforge and setting everything up again using miniconda if you think we should test this.

**Comment by SeaCelo** [comment]
We've tested the process of installing OG-ZAF on a "clean" Mac with the M1 chip. This computer had no previous installation of python, Anaconda, or dependencies. We tried to follow the installation instructions as written in the [Read Me](https://github.com/EAPD-DRB/OG-ZAF). During the installation, we ran into a few problems: 

- Late in the process an error required that users install Xcode Tools. Users should do this as a first step: `xcode-select —install`
- Users should download the correct version of Anaconda. The prominent link in the Anaconda page downloads the X86 version. Instead, users should go to the bottom of the installation page: [Go here](https://www.anaconda.com/products/distribution#Downloads). Then select the version for the M1 (ARM) instead of x86 (Intel). You can check the version of anaconda installed in your system using the command `conda info` and look for "Platform". It should say "osx-arm64".
- With Anaconda for M1 chips installed, you will get an error when you try `conda env create -f environment.yml`. This is because the "MKL" library will not be found (it is an Intel-only library). [To fix this you must follow the instructions in this comment](https://github.com/EAPD-DRB/OG-ZAF/issues/3#issue-1410280233). Basically, avoid installing the "MKL" library.
- Numpy for M1 Macs uses a different acceleration library that is native to ARM processors. We will run some speed tests tomorrow to verify a proposed optimization. I will follow up with the results and the proposed solution. 

Importantly, users with M1 Macs will want to install the appropriate version because there are significant performance gains (we are quantifying this and will provide some data). 

Because of this platform-specific issue, the installation process should be revised to be both universal (Windows, Intel Mac, ARM Macs) and to use the correct version for each platform so users have maximum performance. I'll open an issue to revise the installation instructions and to revise the installation scripts.  

@rickecon

**Comment by SeaCelo** [comment]
A quick update: while I was focused on the acceleration library, the real issue seems to be the Python build. Using the instructions as written you will download the x86 version of Anaconda for Mac, which will install the x86 version of Python 3.9.13. This is a much slower version of Python. Installing the correct ARM version of Anaconda and Python results in a 1.6-1.7x acceleration. 
Issue #15 was opened to update the instructions to reflect this. @rickecon, can we merge these two issues?

**Comment by rickecon** [comment]
This issue is continuing in Issue #15. Closing this issue now.
